### ISSUE TITLE
Sessions not found after updating to a new Zellij version 

### ISSUE THREAD

**Issue by merikan** [zellij-org/zellij#3371]
```
$ zellij --version
zellij 0.40.1
$ alacritty --version
alacritty 0.13.2 (bb8ea18)
$ uname
Darwin
$ sw_vers
ProductName:		macOS
ProductVersion:		14.3.1
BuildVersion:		23D60
```

## Issue description
I accidentally closed a terminal window with a session (named forms). I tried to attach to the session again but it did not work and I get the error "No session with the name 'forms' found!"
```
$ zellij a forms
No session with the name 'forms' found!
```
When I try to list sessions I get the message "No active zellij sessions found."
```
$ zellij list-sessions
No active zellij sessions found.
```
I then tried to start a new session with the name `forms`
```
$ zellij -s forms
```
but the resurrection didn't work (had the same problem with resurrection when going from 0.3x to 0.4x )

When I `ps` zellij I can see the sessions
```
$ ps aux | grep -i zellij
peter            11642   0,4  0,2 410557984 142944   ??  S    Mån08am  65:21.66 /opt/homebrew/bin/zellij --server /var/folders/fw/nhbvnpfx55sbksfzywrfdb_h0000gn/T/zellij-501/0.40.0/personal
peter             4049   0,1  0,3 410650880 199056   ??  S     2Maj24 138:38.61 /opt/homebrew/bin/zellij --server /var/folders/fw/nhbvnpfx55sbksfzywrfdb_h0000gn/T/zellij-501/0.40.0/forms
peter             3861   0,0  0,5 410347152 309472   ??  S     8:50am   0:05.10 /opt/homebrew/bin/zellij --server /var/folders/fw/nhbvnpfx55sbksfzywrfdb_h0000gn/T/zellij-501/0.40.1/forms
```
I can see the old sessions with version `0.40.0` and the newly created with version `0.40.1`

I then downloaded version `0.40.0`
```
~/Downloads
$ ./zellij --version
zellij 0.40.0
```
and killed the newly created session
```
$ zellij delete-session forms --force
A session by the name "forms" exists and is active, but will be force killed and deleted.
Session: "forms" successfully deleted.
```
and then list sessions with the version `0.40.0` binary
```
~/Downloads
$ ./zellij list-sessions
bootcamp [Created 5months 18days 9h 48m 20s ago] (EXITED - attach to resurrect)
temp [Created 1month 2days 9h 26m ago] (EXITED - attach to resurrect)
forms [Created 20days 23h 21m 4s ago]
personal [Created 3days 1h 7m 43s ago]
```
I can now attach to my session `forms` with the version `0.40.0` binary
```
$ ./zellij a forms
```
### Solution
I relied on resurrection and copied directory 0.40.0 to directory 0.40.1 in the cache folder. Kill all running sessions before the copying. 
```
ps aux | grep -i "[z]ellij --server" | awk {'print $2'} | xargs kill
cp -r ~/Library/Caches/org.Zellij-Contributors.Zellij/{0.40.0/,0.40.1}
```
and after that Zellij v0.40.1 recognized my session and could resurrect them 

## Other relevant information
I had similar problem with resurrection when going from `0.3x` to `0.4x` and I had to rename the resurrection folder (if I remember correctly) 

## Proposed solution
Zellij should not use directories with version numbers since it creates all sorts of problems. 
I use long running sessions (months) so I can sometimes do multiple Zellij updates while these sessions are still alive. If I list my sessions (zellij ls), it is expected that all of them are listed and possible to attach to.

If you then shut down one or more sessions, you also expect to be able to resurrect them, without copy folders.

**Comment by iFreilicht** [comment]
The sessions are stored in version-specific folders in the cache (`~/.cache/zellij/` on Linux and `~/Library/Caches/org.Zellij-Contributors.Zellij` on macOS`). If you `cd` into those directories, you'll likely find a `0.40.0` directory. Just copy it to `0.40.1` and all your sessions should show up again.

**Comment by amirkhaki** [comment]
@merikan how did you solved the problem of resurrection? 
i currently have the problem and i don't know how to solve it
i've removed the cache folder completely but no luck

**Comment by merikan** [comment]
@amirkhaki Yes I just copied the old folder (previous version) to a new one with the new version.

**Comment by Dialga** [comment]
A bit unusual to ignore old sessions with every new release isn't it?

**Comment by cbrake** [comment]
still doing this on every upgrade -- is there a reason Zellij can't do this for us?

**Comment by noahbaculi** [comment]
This is just a minor annoyance for me personally. Unfortunately, I can't recommend zellij to my friends and coworkers without this being fixed. The one coworker that tried zellij said this behavior makes zellij feel "unpolished, not ready for daily use", which breaks my heart. I would love to be able to share my love of zellij more widely without needing a disclaimer about upgrades!

**Comment by merikan** [comment]
and here we go again 😀

Today I upgraded from `0.41.2` to `0.42.1`.

1. No actiive sessions
```bash
$ zellij list-sessions
No active zellij sessions found.
```
2. List running zellij processes. We can see the `0.41.2` version.
```bash
$ ps aux | grep -i "[z]ellij --server"
peter            34580  25,3  0,2 538503280 109792   ??  S    20Mar25 453:39.59 /opt/homebrew/bin/zellij --server /var/folders/fw/nhbvnpfx55sbksfzywrfdb_h0000gn/T/zellij-501/0.41.2/bootcamp
peter            72586  21,0  0,3 505703104 196400   ??  S    14Mar25 497:53.38 /opt/homebrew/bin/zellij --server /var/folders/fw/nhbvnpfx55sbksfzywrfdb_h0000gn/T/zellij-501/0.41.2/forms
peter            70405  19,3  0,2 496488160 135232   ??  S    13Mar25 449:53.92 /opt/homebrew/bin/zellij --server /var/folders/fw/nhbvnpfx55sbksfzywrfdb_h0000gn/T/zellij-501/0.41.2/personal
``` 
3. Kill all running sessions
```bash
ps aux | grep -i "[z]ellij --server" | awk {'print $2'} | xargs kill
```
4. Copy the content of the resurrection directory, from `0.41.2` to `0.42.1`
```bash
cp -r ~/Library/Caches/org.Zellij-Contributors.Zellij/{0.41.2/,0.42.1}
```
5. and list sessions
```bash
$ zellij ls
sub-shell [Created 14m 48s ago] (EXITED - attach to resurrect)
nix [Created 14m 48s ago] (EXITED - attach to resurrect)
personal [Created 14m 48s ago] (EXITED - attach to resurrect)
test [Created 14m 48s ago] (EXITED - attach to resurrect)
lab [Created 14m 48s ago] (EXITED - attach to resurrect)
forms [Created 14m 48s ago] (EXITED - attach to resurrect)
bootcamp [Created 14m 48s ago] (EXITED - attach to resurrect)
```

**Comment by Litarvan** [comment]
We install Zellij using repository-level nix shells, meaning the version can change pretty often when switching between branches. Any Zellij update creates disturbances for everyone during a few weeks :( It's not that much, zellij is still extremely useful, but it would be nice to have a workaround!
I guess the reason the sessions are separated by version is to prevent any compatibility issue, but I think having at least a way to list and kill the old ones would be pretty useful, would you accept a PR for that?

**Comment by airtonix** [comment]
got bit by this today i thought zellij was broken . 

we really need a `zellij migrate-sessions --from 0.43.1` 

even if someone makes a wasm plugin for this. don't care

**Comment by pilgrimlyieu** [comment]
After upgrading to 0.44.0, the previous manual migration methods no longer seem to work. However, I noticed that moving the `session_info` from the old version into `contract_version_1` can restore it, i.e.:

```bash
$ cp -r ~/.cache/zellij/0.43.1/session_info ~/.cache/zellij/contract_version_1/
```

> [!NOTE]
> **[Forwards Compatibility](https://zellij.dev/news/remote-sessions-windows-cli/#forwards-compatibility) in release note**
> 
> Due to the lack of a client-server contract, Zellij sessions have never been backwards compatible. Each version upgrade would orphan existing sessions, forcing users to manually kill them and recreate them. While this is unfortunately still the case with this version, this will be the last time.
>
> A new client/server contract has been created and enforced with protocol buffers. Future versions will be able to connect to existing sessions. The only caveat being that new features won’t work with old sessions and will silently fail.

**Comment by airtonix** [comment]
@pilgrimlyieu thanks this fixed it for me

**Comment by cnwangjie** [comment]
> After upgrading to 0.44.0, the previous manual migration methods no longer seem to work. However, I noticed that moving the `session_info` from the old version into `contract_version_1` can restore it, i.e.:
> 
> $ cp -r ~/.cache/zellij/0.43.1/session_info ~/.cache/zellij/contract_version_1/

You really saved my life. Also post a copy-as-you-go code for MacOS users:

```bash
cd ~/Library/Caches/org.Zellij-Contributors.Zellij
cp -r 0.43.1/session_info contract_version_1/
```
